File: 2260-minimum-consecutive-cards-to-pick-up/2260-minimum-consecutive-cards-to-pick-up.java
class Solution {
    public int minimumCardPickup(int[] cards) {
        int n = cards.length;
        HashSet<Integer> set = new HashSet<>();
        
        int minCards = n + 1;
        int i = 0, j = 0;
        while(j < n) {
            if(set.contains(cards[j])) {
                minCards = Math.min(minCards, set.size() + 1);
                set.remove(cards[i++]);
            } else {
                set.add(cards[j++]);
            }
        } 
        
        return (minCards == n + 1) ? -1 : minCards;
    }
}

File: 2260-minimum-consecutive-cards-to-pick-up/2260-minimum-consecutive-cards-to-pick-up.cpp
class Solution {
public:
    int minimumCardPickup(vector<int>& cards) {
        int n = cards.size();
        
        unordered_map<int,int> mp;
        int minCards = n + 1;
        
        for(int i = 0; i < n; i++) {
            if(mp.find(cards[i]) != mp.end()) {
                minCards = min(minCards, i - mp[cards[i]] + 1);
            }
            
            mp[cards[i]] = i;
        }
        
        return (minCards == n + 1) ? -1 : minCards;
    }
};